### DISCUSSION TITLE
Pdf export hatch_policy configuration parameter

### DISCUSSION THREAD

**Discussion by maxm199** [mozman/ezdxf#1130] [Q&A]
Hi all, I'm having an issue with exporting as pdf a drawing with a hatch.
When I try to export as pdf the dxf I created, either with matpolotlib or pymupdf, with the configuration parameter hatch_policy=HatchPolicy.NORMAL it takes literally forever without creating the pdf file.
If I put a timeout of 10 seconds with hatching_timeout=10 then an alert comes out saying:
hatching timeout of 10s reached for HATCH(#None) - aborting
And then nothing else.
I tried to let it go for as long as 30 minutes without having a pdf.
No issues (less than 2 seconds to generate the pdf) if I put hatch_policy=HatchPolicy.SHOW_OUTLINE

This is the drawing, the hatch is the wall filling, nothing else.

![image](https://github.com/user-attachments/assets/86dfc9df-6010-4c3a-a23d-a0f60cf957b9)

**Comment by maxm199** [comment]
I've done some testing and I think I found the issue.
Altering the hatch scale in the dxf drawing I've been able to produce a pdf, it took few minutes anyway.

This is the dxf hatch as it appear in the viewer.

![image](https://github.com/user-attachments/assets/59da108f-92bd-43ae-89b5-1bb41c9f83f0)

And this is the pdf hatch, it seems it misses a hatch scaling parameter, so the hatch is much denser than the dxf one.

![image](https://github.com/user-attachments/assets/eb83f4fd-3caf-426b-9d2c-7033c5268f90)

Is there a way to include a parameter for scaling the hatch?

**Comment by maxm199** [comment]
After more testing and looking trough the code I found a solution.

In the frontend.py file I added a line:

```
    def draw_hatch_pattern(
        self, polygon: DXFPolygon, paths: list[Path], properties: Properties
    ):
        if polygon.pattern is None or len(polygon.pattern.lines) == 0:
            return
        # added to keep the hatch scale same as the dxf drawing
        polygon.set_pattern_scale(1.0)
        ocs = polygon.ocs()
        elevation = polygon.dxf.elevation.z
        properties.linetype_pattern = tuple()
        lines: list[tuple[Vec3, Vec3]] = []
```

I can't see any issues with this change in the code.

**Comment by mozman** [comment]
Hi!

Scaling the patterns works so far for all drawings I've tested and apparently for other users of the library as well.

Therefore, your drawing is the exception and there must be a good reason and explanation why I should change the code of a stable part of the library.

**Comment by maxm199** [comment]
Hi, maybe I did not explain well my situation.

The pattern scaling works fine in the dxf, is when I make the pdf that the pattern comes out with a strange scaling, as you see in my second post I would have liked the pdf hatch (white background) to be the same as the dxf hatch (black background), I did not find a scaling attribute for the hatch in the configuration of the drawing addon, but maybe I missed it, if you can point it out for me I will definitely use that instead.

Setting the scale to 1 in that point did not change the scaling in the dxf drawing (I can still change it as I like) but made the pattern in the pdf the same scale as the dxf.
So now if I change the scaling in the dxf hatch the pdf hatch comes out with the same scale.

DXF
![image](https://github.com/user-attachments/assets/01d970b1-1f5f-4b4a-aab1-83c7e11a3835)

PDF
![image](https://github.com/user-attachments/assets/6755aef2-fe57-4812-a024-ed78f55f9c45)


I hope to have explained better the situation.

**Reply by mozman** [reply]
I still don't understand you issue, but there is no overall scaling factor for HATCH entities. You can of course scale any HATCH entity individually before the PDF export (don't save the modified version).

**Comment by maxm199** [comment]
My issue is that I put a pattern inside my dxf, scale it so it displays as I like and when I make the pdf I have a different pattern result and I don't understand why.

My dxf is the A, with the pattern "GRAVEL" scaled to 50, I'm expecting that when I make the pdf it comes out like B, instead I have C and I don't understand why.
Putting polygon.set_pattern_scale(1.0) into that function makes the pdf come out as B (scaled to 50?), as I expected initially, I don't know why.
The same happens if I put any other pattern with any other scale in the dxf.
All started because I could not produce a pdf with some combination of pattern/scale and I think this was happening because there were too much details coming out in the pdf, while in the dxf I saw just a few.

Changing the scale before making the pdf could be a solution but change it to what?
If I don't understand why and how this pattern changes are happening I cannot have a repeatable behavior.

![image](https://github.com/user-attachments/assets/8758c46a-454c-4837-a41a-052dcdecf4b1)
